sdk/helper/scaleutils/nodeselector/defualt.go

```diff
@@ -0,0 +1,36 @@
+package nodeselector
+
+import (
+	"github.com/hashicorp/nomad-autoscaler/sdk"
+	"github.com/hashicorp/nomad/api"
+)
+
+// defaultClusterScaleInNodeSelector is the NodeSelector implementation of the
+// ClusterScaleInNodeSelector interface. It selects nodes based off the list
+// ordering from Nomad and is the default selector.
+type defaultClusterScaleInNodeSelector struct{}
+
+// newDefaultClusterScaleInNodeSelector returns a new
+// defaultClusterScaleInNodeSelector implementation of the
+// ClusterScaleInNodeSelector interface.
+func newDefaultClusterScaleInNodeSelector() ClusterScaleInNodeSelector {
+	return &defaultClusterScaleInNodeSelector{}
+}
+
+// Name satisfies the Name function on the ClusterScaleInNodeSelector
+// interface.
+func (d *defaultClusterScaleInNodeSelector) Name() string {
+	return sdk.TargetNodeSelectorStrategyDefault
+}
+
+// Select satisfies the Select function on the ClusterScaleInNodeSelector
+// interface.
+func (d *defaultClusterScaleInNodeSelector) Select(nodes []*api.NodeListStub, num int) []*api.NodeListStub {
+
+	out := make([]*api.NodeListStub, num)
+
+	for i := 0; i <= num-1; i++ {
```

Comment:
what if `len(nodes) < num`?

not sure if earlier code protects, but still probably worth handling.

Reply:
Good shout. It's protected from the current caller, but this doesn't protect future implementations.